src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,53 @@
+package lotto.domain;
+
+import java.util.Arrays;
+
+
+public enum Rank {
+
+    FIRST(6, 200000000),
+    SECOND(5, 30000000),
+    THIRD(5, 1500000),
+    FOURTH(4, 50000),
+    FIFTH(3, 5000),
+    NOTHING(0,0);
+
+
+    private final int matchCount;
+    private final Money prize;
+
+
+    Rank(final int matchCount,final int prize) {
+        this.matchCount = matchCount;
+        this.prize = new Money(prize);
+    }
+
+    public static Rank countMatch(final Winning winning, final Lotto lotto){
+        int lottoCount = (int) winning.winningLottoNumber(lotto);
+        boolean bonusBallMatch = winning.bonusLottoNumber(lotto);
+        if (THIRD.match(lottoCount) && !bonusBallMatch) {
+            return THIRD;
+        }
+        return Arrays.stream(Rank.values()).filter(rank -> rank.match(lottoCount)).findFirst()
```

Comment:
stream은 한 줄에 하나의 메서드만 호출하도록 해주세요! 😃
이렇게 사용하면 인텔리제이가 각 줄에서 어떤 결과 타입이 반환되는지 확인할 수 있어요! (아래의 그림과 같이 나옵니다!🤗)
```suggestion
        return Arrays.stream(Rank.values())
                       .filter(rank -> rank.match(lottoCount))
                       .findFirst()
```

![image](https://user-images.githubusercontent.com/47850258/152648251-435aa9d1-57cf-4b83-b876-e3db1b333d91.png)